examples/nlp/token_classification/punctuation_capitalization.py

```diff
@@ -52,12 +52,9 @@ def main(cfg: DictConfig) -> None:
         model = PunctuationCapitalizationModel.from_pretrained(cfg.pretrained_model)
         data_dir = cfg.model.dataset.get('data_dir', None)
         if data_dir:
-            # we can also do finetunining of the pretrained model but it will require
-            # setting up train and validation Pytorch DataLoaders
-            model.setup_training_data(data_dir=data_dir)
-            # evaluation could be done on multiple files, use model.validation_ds.ds_items to specify multiple
-            # data directories if needed
-            model.setup_validation_data(data_dirs=data_dir)
+            model.update_data_dir(data_dir)
+            model.setup_training_data()
+            model.setup_validation_data()
             logging.info(f'Using config file of the pretrained model')
```

Comment:
I really do not understand this whole construct...

The `setup_*_data` methods have `cfg` parameter that should be set in order to "reconfigure"/set new dataset/dataloader. Why not to simply use it?

Do we really need a separate method "overriding" data dir of model.cfg field... that is read from cfg.model.dataset?

Reply:
In this example, we're reusing a pre-trained model and the only thing in the pretrained model's config we need to change is the data_dir defined in `model.cfg.dataset.data_dir` - using `update_data_dir()` for that. We can use `cfg` parameter in `setup_*_data`, but for this particular case, that is not needed.